src/serializers/filter.rs

```diff
@@ -163,15 +173,23 @@ trait FilterLogic<T: Eq + Copy> {
                     return Ok(None);
                 }
             } else if let Ok(include_set) = include.downcast::<PySet>() {
-                if include_set.contains(py_key)? || include_set.contains(intern!(include_set.py(), "__all__"))? {
+                if include_set.contains(py_key)? || include_set.contains(intern!(py, "__all__"))? {
+                    return Ok(Some((None, next_exclude)));
+                } else if !self.explicit_include(int_key) {
+                    // if the index is not in include, include exists, AND it's not in schema include,
+                    // this index should be omitted
+                    return Ok(None);
+                }
+            } else if let Ok(contains_method) = include.getattr(intern!(py, "__contains__")) && let Ok(result) = contains_method.call1((py_key.to_object(py),)) {
```

Comment:
This allowed me to, with no performance penalty as far as I can see, not raise an error if `__contains__` is not callable (which could be done on purpose to prevent the `x in y` check from being possible even if `y` is iterable).